src/components/Card.js

```diff
@@ -1,30 +1,51 @@
 export default class Card {
-  constructor({ name, link }, cardTemplate, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardTemplate,
+    handleImageClick,
+    handleDeleteClick,
+    handleAddLike
+  ) {
     this.name = name;
     this.link = link;
+    this._id = _id;
+    this.isLiked = isLiked;
     this._cardTemplate = cardTemplate;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleAddLike = handleAddLike;
   }
 
   _setEventListeners() {
     this._likeBtn.addEventListener("click", () => {
-      this._handleLikeIcon();
+      this._handleAddLike(this);
     });
 
-    this._trashBtn.addEventListener("click", () => {
-      this._handleDeleteCard();
-    });
+    this._cardElement
+      .querySelector("#trash-button")
+      .addEventListener("click", () => {
+        this._handleDeleteClick(this);
+      });
 
     this._cardImageEl.addEventListener("click", () => {
       this._handleImageClick({ name: this.name, link: this.link });
     });
   }
 
-  _handleLikeIcon() {
-    this._likeBtn.classList.toggle("card__like-button_active");
```

Comment:
Needs correction: please rename ` handleLikeIcon` to something like `setIsLiked`, since it's not an event handler anymore